### PR TITLE
Feat/integracao api jwt e api salgado

### PR DESCRIPTION
**Pull request by DanielRRSilva**

### CODE REVIEW

File: jwt-api/src/main/resources/application.properties
Comment (davifeitosa0) [inline jwt-api/src/main/resources/application.properties:1]: senha daora 😃

Comment (davifeitosa0) [reply jwt-api/src/main/resources/application.properties:1]: põe em env

**Comment by julia-hikari** [comment]
[like]  JULIA HIKARI KUNIHOSITI KURAHA... reacted to your message:
________________________________

**Comment by julia-hikari** [comment]
[like]  JULIA HIKARI KUNIHOSITI KURAHA... reacted to your message:
________________________________
From: Daniel Ricardo Rosa e Silva ***@***.***>
Sent: Thursday, October 31, 2024 12:15:27 PM
To: Origami-Flow/origami.flow.api ***@***.***>
Cc: JULIA HIKARI KUNIHOSITI KURAHARA . ***@***.***>; Review requested ***@***.***>
Subject: [Origami-Flow/origami.flow.api] Feat/integracao api jwt e api salgado (PR #5)

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/Origami-Flow/origami.flow.api/pull/5

Commit Summary

  *   1f7e780<https://github.com/Origami-Flow/origami.flow.api/pull/5/commits/1f7e780d7a391a8f2e0e92f1639eebaf19e64d29> integracao parcialmente feita, criptografia da senha no cadastro e geracao de token funcionando
  *   e3bbb0e<https://github.com/Origami-Flow/origami.flow.api/pull/5/commits/e3bbb0e72bc7ead94c926969d7410e1038997a52> validacao toke jwt funcionando

File Changes

(69 files<https://github.com/Origami-Flow/origami.flow.api/pull/5/files>)

  *   D demo/.gitignore<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-18fa4d4cbe98b740aa6f62b1d622d1eb1ccb49e5c62383c112a2ca8c7db7950d> (38)
  *   D demo/.idea/.gitignore<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-9b3561738891342574b9a21514615e2c4982966b46048494e75098d3b5da2fdf> (8)
  *   D demo/.idea/encodings.xml<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-fb9377530377820c5dfb3a0c264a4a6b9016e2972ed2dfdce4d2a03ad77ee305> (7)
  *   D demo/.idea/misc.xml<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-65e4d4f72e1cc2b1dbfc2ae13d4d8c47de7093fd0dda0c46c2494e56c3f13a71> (15)
  *   D demo/.idea/uiDesigner.xml<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-8b9a44013b58122082424138bd19eb57d07daa3924ae4227fc27a6edb550e613> (124)
  *   D demo/.idea/vcs.xml<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-746896a14062438271af0ade4859a6cc430e4456031168190ede615dedaae6b6> (6)
  *   D demo/.mvn/wrapper/maven-wrapper.properties<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-d98002ac8ba118279cefa7d5bb72e287cb62bd5d1e96be45c00be7d729fa56cf> (19)
  *   D demo/HELP.md<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-bacc7928624ea8bd4768678644836424e267b2a835c1a7e3f573cfb5718c2a80> (21)
  *   D demo/mvnw<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-b603eced56a43efeb9f740e01058e6a32a34c7b226a8b1cc6d6b6719fabab02b> (259)
  *   D demo/mvnw.cmd<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-b0e6b286062d5c707813d42a3e6362faf3dcb53bfa97be32aa77ab8e5f7a3e3e> (149)
  *   D demo/pom.xml<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-00a72cc680d374a97c714076c99acc0075788972a1452e3a5630dac8129cb6e3> (145)
  *   D demo/src/main/java/api_jwt/demo/DemoApplication.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-de0f2aec641921778d1a67230a04dfed4e16ff6ea5f61cffd26b96dcd653bfda> (13)
  *   D demo/src/main/java/jwt/api/api/controller/usuario/UsuarioController.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-472b6ab43589840aefa9c8685274fd73ba41c99174c3425befa6825ed13a5404> (35)
  *   D demo/src/main/java/jwt/api/configuration/security/AutenticacaoEntryPoint.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-791e839b1007dc6ac4f98d8c094cff75125d37b4658acba0e9caa598711b6bde> (24)
  *   D demo/src/main/java/jwt/api/configuration/security/AutenticacaoFilter.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-1645699a5faf023fce851c40d02ec5da063b0d71fa59fc41b07765c66977610b> (75)
  *   D demo/src/main/java/jwt/api/configuration/security/AutenticacaoProvider.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-8f8f67eb5cf7cc9cabf810d257346d87f38a826c118a3dfa261012c2a28932cb> (40)
  *   D demo/src/main/java/jwt/api/configuration/security/SecurityConfiguracao.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-d251cf54d83833d652b7bbae883c6c0e91ce98da490400e4d854362540cd4338> (139)
  *   D demo/src/main/java/jwt/api/configuration/security/jwt/GerenciadorTokenJwt.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-e03b28634dfb123b3cb45ac63c51a30990004a97a50976415bb7c66eb8c7acc2> (70)
  *   D demo/src/main/java/jwt/api/demo/DemoApplication.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-c5cd44b66fa783443b931cd8872e1c48b5e8e76d4952f75d150874fecb10e712> (13)
  *   D demo/src/main/java/jwt/api/mapper/UsuarioMapper.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-ce966c73bbfbcdf360da8eb53755600ccfcdad86d7473dd8b00a843aea6f10dc> (14)
  *   D demo/src/main/java/jwt/api/pacote/Main.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-cca9131bc7f40f759d14fc8007c6244c4b5bd7f56a7847af9769608b6abfe97b> (13)
  *   D demo/src/main/java/jwt/api/service/usuario/UsuarioService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-b41f6ed51ad8c4fef9c313cd762208dc442f8c7303a6567f092fc7e924deb623> (54)
  *   D demo/src/main/java/jwt/api/service/usuario/autenticacao/AutenticacaoService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-98b9259fd373280488f19182ab7de1ae2cfb7d059c7c7c93b2a844a78bb4f5e3> (27)
  *   D demo/src/main/java/jwt/api/service/usuario/autenticacao/dto/UsuarioDetalhesDto.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-77c0768602d3f11d4431ba20ca62b1795558164f500f73461934e4b64e8ee1cd> (58)
  *   D demo/src/main/java/jwt/api/service/usuario/autenticacao/dto/UsuarioLoginDto.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-1c1e0ee96c01def08d0863af137c0dc3ee199bb628156f1c83b5e41060913590> (12)
  *   D demo/src/main/java/jwt/api/service/usuario/dto/UsuarioCriacaoDto.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-cd62a1ed1fe7cbd68c46c95e1710ce5fe62e60063d6db08ee8edb8b9457e0ad8> (28)
  *   D demo/src/main/resources/application.properties<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-d8fdad051e8c6abebcb0fe14265d5cb278b43bc813ebc2c6381915626a56514e> (28)
  *   D demo/src/main/resources/data.sql<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-63f6d7990dd1aba97a850bd26a02b3e16d54d1c0c334f14325963a7829e016ba> (2)
  *   D demo/src/test/java/api_jwt/demo/DemoApplicationTests.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-7b66f79b9fe50d359853101dc4c3de2569c588577492fdd32e82c18f966f8fcc> (13)
  *   M jwt-api/src/main/java/jwt/api/configuration/security/SecurityConfiguracao.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-9de4aa9b456871673dbf0472f146c26d963487d050bfb54add5e97a7484f257d> (5)
  *   R jwt-api/src/main/java/jwt/api/domain/Cliente.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-408ddc2084a51785da9c296bdc4ac43e8041fd15575abb022a1b9d2f8b1880ea> (16)
  *   A jwt-api/src/main/java/jwt/api/domain/Trancista.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-dea5dfd834e75219a52b196f31d0f90b4ed3b9181d8b71daf73f1b6268d33cb3> (30)
  *   D jwt-api/src/main/java/jwt/api/domain/Usuario.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-afd17d03bbce80e41f3317a83f15f420eebcb26af724be2eeff92a888f8382e4> (50)
  *   R jwt-api/src/main/java/jwt/api/domain/repository/ClienteRepository.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-3d01723820e894f907455bff8b998344abf9b9be10bb28503f61e4a2c764700c> (6)
  *   R jwt-api/src/main/java/jwt/api/domain/repository/TrancistaRepository.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-9e5d4858e1e0701bac4ae003b08eeb5c67c8c4241709b0d5617487966d0efcef> (7)
  *   A jwt-api/src/main/java/jwt/api/service/usuario/AutenticationService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-f0ce531d465e124085d6a83afd5b70756bd9c66103840a81429fefd68705a87e> (63)
  *   D jwt-api/src/main/java/jwt/api/service/usuario/UsuarioService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-725538c21c5fd8e1a6d607cd178003622df5b22faa70af91c4009be9516194a0> (64)
  *   M jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/AutenticacaoService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-7416de7b0977680612fea0bfacb5067b794e019aca703cc87a24915326364518> (9)
  *   A jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/dto/CadastroCriptografado.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-59851f1e24580415bbebcbd6f1956fed2880dfd098af170b2701d756835b4a54> (19)
  *   A jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/dto/CadastroRequestDTO.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-378b7d5aa8a6b0caec3e226021b4abe62e80bdaf9784c98b315efa7f94ceabf0> (25)
  *   R jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/dto/JwtTokenResponse.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-aa1af129f86d9b0bc4f3d53ec764171beac20a5870ae990f774060e5a1984988> (16)
  *   R jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/dto/LoginRequestDTO.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-1993491abb6fc7f957d28eb4ecc40e353c50b04c8381dd55c8ac9fec1d13e7d1> (22)
  *   M jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/dto/UsuarioDetalhesDto.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-d6fbacad1c061d3c17eec73068800c06a1a7e5572d83a3210394e68770883c37> (10)
  *   D jwt-api/src/main/java/jwt/api/service/usuario/autenticacao/dto/UsuarioTokenDto.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-719beb636127ab86c94e51530f30aec292202d768f0a8625343251ea7c91c87e> (54)
  *   D jwt-api/src/main/java/jwt/api/service/usuario/dto/UsuarioCriacaoDto.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-cafe9d45077833231703b66241101dc62c73053e5a8ee07c8c12d4ee9cbe40ea> (48)
  *   D jwt-api/src/main/java/jwt/api/service/usuario/dto/UsuarioMapper.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-a4ded2c24e3ff9786b176244828c120b66f0d58a9e6d51dd1c3f5ff85dbe1e20> (29)
  *   A jwt-api/src/main/java/jwt/api/service/usuario/mapper/CadastroCriptografadoMapper.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-acc87779dfc52f82ac6ff93aac283d6c9f2e7da09680e2aba2e2bb65ec6423bb> (13)
  *   A jwt-api/src/main/java/jwt/api/service/usuario/mapper/UsuarioMapper.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-7c5622257a64a126fd46f9c6461fc82bc8aaa390ebb38f3ba88bd31b7fbe3351> (40)
  *   A jwt-api/src/main/java/jwt/api/uso/controller/usuario/AutenticationController.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-88dd253a5f745b06ee8c92f017fcdacb6fcd01c4a5aab74d17264e143199ebd1> (45)
  *   D jwt-api/src/main/java/jwt/api/uso/controller/usuario/UsuarioController.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-1c78e0bca8d20a10dfd62b3073f919dbd2f28381395b188459fe01da9370b1c5> (36)
  *   M jwt-api/src/main/resources/application.properties<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-efb19dc000dcb71e1d4ee50ac95767068e180dd1eabf7966dd454ae42786fb2d> (8)
  *   M salgado-trancas-api/pom.xml<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-9e28686d15a8a7c880675113009e15399eb319b4fd477c36c11e253a8a0a7eea> (32)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/configs/SecurityConfig.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-eb80297919d098cdbaa47b945f82fcef011078cb149f76934916c261d6c838f9> (29)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/controller/CadastroController.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-c1436187b09abab844d71ee3ad2df6f7c5825354c84b1401ba99783476205f4c> (4)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/controller/LoginController.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-b68d6cd25af8b63b900a8cce893df7417fe68a1d52501aaf38605e9ef93cd5be> (21)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/dto/request/CadastroRequestDTO.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-e8eb87490d3337ca246f3eb4f1f7c4ed0f2e29ccb17710cf5b8bc66ff78e5b70> (5)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/dto/request/LoginRequestDTO.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-7794f70b3eafcf38f88358d180f03e54d7231f94fc47095a63873b96697b6f58> (36)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/dto/response/JwtTokenResponse.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-3ed9d20b4018b052c371cafcca041774c1705d39fe095c474b35bcd8df60272d> (21)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/dto/response/cadastro/CadastroCriptografadoResponse.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-ff45cb1a139e50d0007957e9005302af7fa34202f7dfe2b04dfe2d2acfd07dbf> (20)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/entity/UsuarioAbstract.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-4768d8052edd4b1b04109ae473e4744d69e27c99fc9f490e447542b82bdeae84> (2)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/mapper/CadastroMapper.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-3b449113ef7fb7203ebf7e3b6ed32df72f9fbfaca32a28564267e67962f301d6> (35)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/repository/EnderecoRepository.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-8fef3dbf6ba1237ffdc55b369b8e1f2d4fc5d7f8a9595a57af6896b5b9d5ac2e> (2)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/service/CadastroService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-a5675f0904ad4bebe04a6fb4d210632c6baf8607056f1e67c806cb6f2709aefa> (8)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/service/EnderecoService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-7dd8d16baba2050adc4480240b90297d816a2156d7fb1299311ef4f6ae008d4a> (2)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/service/LoginService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-0db0784ca9a2ded68be7dc56607de778b70f4db2ef37934ae5dfc5c1748e039c> (23)
  *   M salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/service/ProdutoService.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-ee3e110d0e010d883f52c56d4a9e1628b7e7eb0fcf7e8b7b6c6845e44371dd02> (1)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/utils/ConexaoApiJwt.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-318267c22ae30353d4625d0d51243df0b69da1089b3a6655d0fe27761736462a> (57)
  *   A salgado-trancas-api/src/main/java/origami_flow/salgado_trancas_api/utils/JwtFilter.java<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-73fb1f42ea47518031be12c07fe941f2b4c895c8ac5a74d33b04f3d770c317fb> (45)
  *   M salgado-trancas-api/src/main/resources/application.properties<https://github.com/Origami-Flow/origami.flow.api/pull/5/files#diff-9fd16256b090cc75bd4cbe2c41eff64fdecbb8270f40b4d8177bde0751aa2e4a> (20)

Patch Links:

  *   https://github.com/Origami-Flow/origami.flow.api/pull/5.patch
  *   https://github.com/Origami-Flow/origami.flow.api/pull/5.diff

**Pull request merged**